### PR TITLE
Chex CLI for checking DEXPI P&ID

### PR DESCRIPTION
**Pull request by henriettelienrebnor**
Created the `chex` CLI , a command-line tool for validating P&ID files with the essential criteria required for the XSLT and RML mappings in this repository. 

Also updated one of the rml mappings to be more specific. The only elements we want to check previous / following siblings of are of type Node.

### CODE REVIEW

File: cli/chex/chex.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+@cli.command()
+@click.argument("file_path", type=str, nargs=1)
+def shapes(file_path):
+    """Checks how many elements in the ShapeCatalouge have the correct formatting. Also checks how many known and unkown shapes are present."""
+    tree = etree.parse(file_path)
+    check_shapes(tree)
+
+@cli.command()
+@click.argument("file_path", type=str, nargs=1)
+def unkown(file_path):
```
Comment (markus-ap) [inline cli/chex/chex.py:41]:
```suggestion
def unknown(file_path):
```
Just a small typo!

File: cli/chex/chex.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    """Prints all shapes in the P&ID that is known in the NOAKADEXPI symbol library"""
```
Comment (markus-ap) [inline cli/chex/chex.py:42]: To me there seems to be a mismatch between the call being called `unknown` and the documentation saying that it prints all that is `known`.

File: cli/chex/chex.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    tree = etree.parse(file_path)
+    known_shapes = get_known_shapes(tree)
+    print("Known shapes:")
+    print("--------------")
+    for known in known_shapes:
+        print(f"  {known}")
+
+@cli.command()
+@click.argument("file_path", type=str, nargs=1)
+def known(file_path):
```
Comment (markus-ap) [inline cli/chex/chex.py:52]: Same as with the `unknown` method. 
Perhaps it makes sense when one uses it, but typing `known` and getting `unknown` seems counter-intuitive for me.

File: cli/chex/dexpi.xml [hunk 4]
```diff
@@ -0,0 +1,4924 @@
+<?xml version='1.0' encoding='UTF-8'?>
```
Comment (markus-ap) [inline cli/chex/dexpi.xml:1]: Does this file need to be included here? I can't find any usages of it.

**Review by markus-ap — COMMENTED** [review]
Good work! A real useful CLI for checking if a DEXPI XML is compliant for the mappings.

Leaving off an approval until the file `dexpi.xml` has been clarified.

File: cli/chex/chex.py [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+@click.argument("file_path", type=str, nargs=1)
+def shapes(file_path):
+    """Checks how many elements in the ShapeCatalouge have the correct formatting. Also checks how many known and unkown shapes are present."""
+    tree = etree.parse(file_path)
+    check_shapes(tree)
+
+@cli.command()
+@click.argument("file_path", type=str, nargs=1)
+def unknown(file_path):
+    """Prints all shapes in the P&ID that is known in the NOAKADEXPI symbol library"""
```
Comment (henriettelienrebnor) [inline cli/chex/chex.py:42]:
```suggestion
    """Prints all shapes in the P&ID that is unknown in the NOAKADEXPI symbol library"""
```

File: cli/chex/chex.py [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+    known_shapes = get_known_shapes(tree)
+    print("Known shapes:")
+    print("--------------")
+    for known in known_shapes:
+        print(f"  {known}")
+
+@cli.command()
+@click.argument("file_path", type=str, nargs=1)
+def known(file_path):
+    """Prints all shapes in the P&ID that is not known in the NOAKADEXPI symbol library"""
```
Comment (henriettelienrebnor) [inline cli/chex/chex.py:53]:
```suggestion
    """Prints all shapes in the P&ID that is known in the NOAKADEXPI symbol library"""
```

File: cli/chex/chex.py (hunk 2 above)
Comment (henriettelienrebnor) [reply cli/chex/chex.py]: I mixed the documentation for these two functions! Corrected it now

File: cli/chex/dexpi.xml (hunk 4 above)
Comment (henriettelienrebnor) [reply cli/chex/dexpi.xml:1]: I mixed the documentation for these two functions! Corrected it now

Comment (henriettelienrebnor) [reply cli/chex/dexpi.xml:1]: I forgot to remove the file I used for testing the CLI. Will fix asap

File: cli/chex/chex.py [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+@cli.command()
+@click.argument("file_path", type=str, nargs=1)
+def known(file_path):
+    """Prints all shapes in the P&ID that is known in the NOAKADEXPI symbol library"""
+    tree = etree.parse(file_path)
+    unknown_shapes = get_unkown_shapes(tree)
+    print("Unknown shapes:")
+    print("--------------")
+    for unknown in unknown_shapes:
+        print(f"  {unknown}")
```
Comment (eoye) [inline cli/chex/chex.py:59]: I think there is a little mixup now with "uknown" printing known shapes and vice versa, but other than that it looks good!

**Review by eoye — APPROVED** [review]
Nice work, I tested the cli and it works great!🙌 one small comment with just a method name mixup

File: cli/chex/chex.py (hunk 7 above)
Comment (henriettelienrebnor) [reply cli/chex/chex.py]: Have fixed this now :D

**Pull request merged**